lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -474,6 +578,37 @@ def create_test_xcconfig_files(test_native_targets, test_resource_bundle_targets
             end
           end
 
+          # Generates the contents of the xcconfig file used for each test target type and saves it to disk.
+          #
+          # @param  [Array<PBXNativeTarget>] test_native_targets
+          #         the test native target to link the xcconfig file into.
+          #
+          # @param  [Hash{String=>Array<PBXNativeTarget>}] test_resource_bundle_targets
+          #         the additional test resource bundle targets to link the xcconfig file into.
+          #
+          # @return [void]
+          #
+          def create_app_xcconfig_files(app_native_targets, app_resource_bundle_targets)
+            target.app_specs.each do |app_spec|
+              spec_consumer = app_spec.consumer(target.platform)
+              path = target.xcconfig_path("#{app_spec.name.split('/')[1..-1].join('-')}")
+              update_changed_file(Target::BuildSettings::PodTargetSettings.new(target, app_spec), path)
```

Comment:
build settings might need to be updated, since it will think this `app_spec` is a test spec (that might break e.g. finding app spec dependencies, or differences in settings between tests and apps)